pkg/operator/oauthapiencryptioncontroller/oauthapiencryptioncontroller.go

```diff
@@ -0,0 +1,129 @@
+	return controllerFactory.ToController(name, eventRecorder.WithComponentSuffix("oauth-apiserver-encryption-cfg-sync-controller"))
+}
+
+// sync starts managing oauth-apiserver encryption config (encryption-config-openshift-oauth-apiserver in openshift-config-managed namespace) as described in https://github.com/openshift/enhancements/blob/master/enhancements/etcd/etcd-encryption-for-separate-oauth-apis.md
+//
+// case 1: if the secret doesn't exist and encryption is on then:
+//         - it will add the secret with the annotation (must be atomic operation) because CAO will start managing its own config iff the secret exist without the annotation
+//
+// case 2: if the secret exists and it is annotated
+//         - it will simply start synchronisation
+//
+// case 3: no-op: when the secret exits but it doesn't have the annotation - that means it was created by CAO in 4.6 and this is downgrade
+// case 4: no-op: when the secret doesn't exist and encryption is off
+//
+// drawbacks:
+// - it will not recover when the annotation was manually removed by a user,
+//   to recover we would have to put a value in the annotation instead and coordinate OAS-A and CAO but then we would have to remember to add it in CAO (4.6) as well
+func (c *oauthAPIServerController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	openshiftAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.targetNamespace))
+	if apierrors.IsNotFound(err) {
+		return nil // case 4: encryption off
+	}
+	if err != nil {
+		return err
+	}
+
+	oauthAPIServerEncryptionCfgName := fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace)
+	oauthAPIServerEncryptionCfg, err := c.secretLister.Get(oauthAPIServerEncryptionCfgName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		// case 1: create with annotation
+		if _, exists := openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]; !exists {
+			return fmt.Errorf("%s/%s doesn't contain the required key %q", openshiftAPIServerEncryptionCfg.Name, openshiftAPIServerEncryptionCfg.Namespace, encryptionconfig.EncryptionConfSecretKey)
+		}
+		encryptionCfg := &corev1.Secret{}
+		encryptionCfg.Name = oauthAPIServerEncryptionCfgName
+		encryptionCfg.Namespace = operatorclient.GlobalMachineSpecifiedConfigNamespace
+		encryptionCfg.Annotations = map[string]string{}
+		encryptionCfg.Annotations[encryptionConfigManagedBy] = encryptionConfigManagedByValue
+		encryptionCfg.Annotations[encryptionstate.KubernetesDescriptionKey] = encryptionstate.KubernetesDescriptionScaryValue
+		encryptionCfg.Data = map[string][]byte{}
+		encryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey] = openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
+		encryptionCfg.Finalizers = []string{encryptionsecret.EncryptionSecretFinalizer}
```

Comment:
is there a controller doing finalizing? Or is it just there to block deletion?

Reply:
I added it to block deletion.